### PR TITLE
[Fleet] Fix tests after changes in API

### PR DESCRIPTION
**Pull request by juliaElastic**
Test fixes after Fleet API changes

Note: changes are not breaking, so it's not required to update the tests now, only when the deprecated old APIs are removed

## What does this PR do?

Fix tests after changes here: https://github.com/elastic/kibana/pull/119494

Related issue: https://github.com/elastic/obs-dc-team/issues/655

## Why is it important?

Tests should be kept up-to-date with latest API.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: e2e/_suites/fleet/README.md
Comment (juliaElastic) [inline e2e/_suites/fleet/README.md:144]: Note: these instructions are not working yet locally, took my best guess

**Comment by juliaElastic** [comment]
@v1v initially the API changes were targeting 8.0, but now might slip into 8.1. So we don't have to backport this pr.

File: e2e/_suites/fleet/README.md
Comment (mdelapenya) [inline e2e/_suites/fleet/README.md:164]: @adam-stokes is this needed? IIRC the Docker provider is still enabled for local development, so no need to run against a remote kibana locally. Please correct me if I'm wrong

**Review by mdelapenya — APPROVED** [review]
@juliaElastic changes are good to me! Great job with the updates!!

I'm only waiting for @adam-stokes' feedback on the docs, so let's wait for his approval first. In the meantime, I'm running these tests locally to verify that

File: internal/kibana/server.go
Comment (mdelapenya) [inline internal/kibana/server.go:77]: @adam-stokes I'm not seeing any usage for the `CreateServiceToken` method. Should we remove it?

**Comment by mdelapenya** [comment]
I was able to run the tests with:

```shell
TAGS="integrations" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE DEVELOPER_MODE=true ELASTIC_APM_ACTIVE=false  make -C e2e/_suites/fleet functional-test
```

And verified the new API calls are performed:

```log
TRAC[2021-12-13T11:09:30+01:00] Kibana API Query                              body="{\"policy_id\":\"be3d0900-5bfc-11ec-a139-49a592f43e42\"}" method=POST url="http://localhost:5601/api/fleet/enrollment_api_keys"
...

TRAC[2021-12-13T11:10:26+01:00] Kibana API Query                              body=null method=DELETE url="http://localhost:5601/api/fleet/enrollment_api_keys/8ae22537-881f-497d-a5c7-7d92c9b7ef82"
...
```

File: e2e/_suites/fleet/README.md
Comment (adam-stokes) [reply e2e/_suites/fleet/README.md:164]: Yea this is needed now, I've got a branch im working on that makes all this seemless

File: internal/kibana/server.go
Comment (adam-stokes) [reply internal/kibana/server.go:77]: We should discuss with agent/fleet team on what tests they would like to see with the ServiceToken usage

**Review by adam-stokes — APPROVED** [review]

**Pull request merged**
